experimental/traffic-portal/src/app/core/statuses/status-details/status-details.component.ts

```diff
@@ -0,0 +1,117 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+import { ActivatedRoute, Router } from '@angular/router';
+import { StatusesModel, StatusesService } from 'src/app/api/statuses.service';
+import { DecisionDialogComponent, DecisionDialogData } from 'src/app/shared/dialogs/decision-dialog/decision-dialog.component';
+
+@Component({
+  selector: 'tp-status-details',
+  templateUrl: './status-details.component.html',
+  styleUrls: ['./status-details.component.scss']
+})
+export class StatusDetailsComponent implements OnInit {
+
+  id: string | null = null;
+  statusDetails: StatusesModel | null = null;
+  statusDetailsForm!: FormGroup;
+  loading = false;
+  submitting = false;
+  submitted = false;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private fb: FormBuilder,
+    private readonly dialog: MatDialog,
+    private statusesService: StatusesService) { }
+
+  ngOnInit(): void {
+    // Form is built here
+    this.statusDetailsForm = this.fb.group({
+      name: ['', Validators.required],
+      description: ['', Validators.required],
+    });
+
+    // Getting id from the route
+    this.id = this.route.snapshot.paramMap.get('id');
+
+    // we check whether params is a number if not we shall assume user wants to add a new status.
+    if (!this.isNew) {
+      this.loading = true;
+      this.statusDetailsForm.addControl('id', new FormControl(''));
+      this.statusDetailsForm.addControl('lastUpdated', new FormControl(''));
+      this.getStatusDetails();
+    }
+  }
+
+  /*
+   * Reloads the servers table data. 
+   * @param id is the id passed in route for this page if this is a edit view.
+  */
+  async getStatusDetails(): Promise<void> {
+    const id = this.id as string
+    this.statusDetails = await this.statusesService.getStatuses(id);
+    const data = {
+      name: this.statusDetails.name,
+      description: this.statusDetails.description,
+      lastUpdated: new Date(),
+      id: this.statusDetails.id
+    } as any;
```

Comment:
Nothing may ever be typed as `any`. Even implicitly, `any` should never be the type of anything.